Bindings/Java/Matlab/Utilities/osimC3Dconverter.m

```diff
@@ -0,0 +1,278 @@
+%                                                                         %
+% Unless required by applicable law or agreed to in writing, software     %
+% distributed under the License is distributed on an "AS IS" BASIS,       %
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or         %
+% implied. See the License for the specific language governing            %
+% permissions and limitations under the License.                          %
+% ----------------------------------------------------------------------- %
+
+% Author: James Dunne
+
+p = inputParser;
+value = 0;
+defaultFilePath = '';
+defaultAxis = '';
+expectedAxis = {'x','y','z'};
+
+addOptional(p,'filepath',defaultFilePath)
+addOptional(p,'value',value,@isnumeric);
+addOptional(p,'axis',defaultAxis,...
+                 @(x) any(validatestring(x,expectedAxis)));             
+parse(p,varargin{:});
+
+filepath = p.Results.filepath;
+axis = p.Results.axis;
+value = p.Results.value;
+
+%% Check for file path
+if isempty(filepath)
+        [filein, path] = uigetfile({'*.c3d','C3D file'}, 'Select C3D data file(s)...', 'MultiSelect', 'on');
+else
+    % If the input path to file is wrong, return exception
+    if exist(filepath,'file') == 0
+            error('file does not exist')
+    else    
+        % If the input path is local (called from current folder), you will
+        % need to set the full path. 
+        [path, filename, ext] = fileparts(filepath);
+        if isempty(path)
+            path = cd;
+        end
+        filein = [filename ext];
+    end
+end
+
+if iscell(filein)
+    nFiles = length(filein);
+    multipleFiles = 1;
+else
+    nFiles = 1;
+    multipleFiles = 0;
+end
+    
+for iFile = 1 : nFiles
+    
+    if multipleFiles == 1
+        % get the file name (with extension)
+        filename = filein{iFile};
+    else 
+        filename = filein;
+    end    
+        
+     % get the fullpath to the file 
+    filepath = fullfile(path,filename);
+    % get the file name (without the extension)
+    [path, filename, ext] = fileparts(filepath);
+
+    %% import java libraries
+    import org.opensim.modeling.*
+
+    %% Use a c3dAdapter to turn read a c3d file
+    adapter = C3DFileAdapter();
+    tables = adapter.read(filepath);
+
+    %% get the marker and force data into OpenSim tables
+    markers = tables.get('markers');
+    forces = tables.get('forces');
+
+    %% Rotate marker and force data
+    if isempty(axis)
+        display('Either no axis or value for rotation has been been set; no rotations will be applied')
+        forces_rot = forces.clone();
+        markers_rot = markers.clone();
+    else
+        display(['Rotating Marker and Force data ' num2str(value) ' about the ' axis ' axis']); 
+        markers_rot = osimRotateTableData(markers, axis, value);
+        forces_rot = osimRotateTableData(forces, axis, value);
+    end
+    
+    %% Print the rotated markers to trc file
```

Comment:
I agree the utility should be straightforward and not have too many steps. But, I do think it could call a separate utility for writing, which could be equally handy. Also OK to address in a future PR.